### PR TITLE
add appengine samples

### PR DESCRIPTION
**Pull request by JustinBeckwith**
This is a collection of nodejs samples for MVM that I originally created over here:
https://github.com/JustinBeckwith/appengine-nodejs-samples

I want to bring all of our samples together under one repository, then I'll delete my original.

### CODE REVIEW

**Comment by jonparrott** [comment]
Isn't socketio still not working? Can we remove that sample for now.

**Comment by JustinBeckwith** [comment]
I had already removed the actual sample, just missed a readme.  It's gone now!

**Comment by jmdobry** [comment]
My manual testing:

- [x] - express
- [x] - geddy
- [ ] - grunt
- [ ] - hapi
- [ ] - koa
- [ ] - kraken
- [ ] - loopback
- [ ] - redis
- [ ] - restify
- [ ] - sails

File: appengine/loopback/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	env_variables:
+  		PORT: 8080
+	```
+
+3. Create a `start` deploy script, and deploy! For convenience, you can modify your `package.json` to use an npm script for deployment:
+
+	```js
+	"scripts": {
+		"start": "node ."
+	    "deploy": "gcloud preview app deploy app.yaml --set-default --project [project id]"
```
Comment (jmdobry) [inline appengine/loopback/README.md]: Not valid JSON, should be:

```json
"start": "node .",
"deploy": "gcloud preview app deploy app.yaml --set-default --project [project id]"

File: appengine/redis/server.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+var redis = require("redis");
```
Comment (jmdobry) [inline appengine/redis/server.js]: This file has an interesting mix of single-quoted and double-quoted strings.

File: appengine/redis/server.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+// read in keys and secrets.  You can store these in a variety of ways.  
+// I like to use a keys.json  file that is in the .gitignore file, 
+// but you can also store them in environment variables
+nconf.argv().env().file('keys.json');
+
+// Connect to a redis server provisioned over at 
+// redisLabs.  See the README for more info. 
+var client = redis.createClient(
+    nconf.get("redisPort"),
```
Comment (jmdobry) [inline appengine/redis/server.js]: I would suggest changing this to `nconf.get("redisPort") || "6379",` so `npm start` will work out of the box with a default redis configuration.

File: appengine/redis/server.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    nconf.get("redisHost"),
```
Comment (jmdobry) [inline appengine/redis/server.js]: I would suggest changing this to `nconf.get("redisHost") || "127.0.0.1",` so `npm start` will work out of the box with a default redis configuration.

File: appengine/redis/server.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+// redisLabs.  See the README for more info. 
+var client = redis.createClient(
+    nconf.get("redisPort"),
+    nconf.get("redisHost"), 
+    {
+        auth_pass: nconf.get("redisKey"), 
+        return_buffers: true
+    }
+).on("error", function (err) {
+    logger.error("ERR:REDIS: " + err);
```
Comment (jmdobry) [inline appengine/redis/server.js]: Running `npm start` without valid (or default, see above comments) redis connection values causes this errback to be called, revealing this error:

```
jdobry-macbookpro:redis jdobry$ npm start

> appengine-redis@1.0.0 start /Users/jdobry/projects/nodejs-docs-samples/appengine/redis
> node server.js

started web process

/Users/jdobry/projects/nodejs-docs-samples/appengine/redis/server.js:21
    logger.error("ERR:REDIS: " + err);
    ^
ReferenceError: logger is not defined
    at RedisClient.<anonymous> (/Users/jdobry/projects/nodejs-docs-samples/appengine/redis/server.js:21:5)
    at RedisClient.emit (events.js:95:17)
    at RedisClient.on_error (/Users/jdobry/projects/nodejs-docs-samples/appengine/redis/node_modules/redis/index.js:196:10)
    at Socket.<anonymous> (/Users/jdobry/projects/nodejs-docs-samples/appengine/redis/node_modules/redis/index.js:106:14)
    at Socket.emit (events.js:95:17)
    at net.js:441:14
    at process._tickCallback (node.js:448:13)
```

**Comment by JustinBeckwith** [comment]
That should do it :)  Once this is merged I'll take a look at some simple tests.

**Comment by jmdobry** [comment]
Closing in favor or #6.

**Pull request closed without merging**